packages/components/src/mobile/audio-player/audio-url-parser.native.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * WordPress dependencies
+ */
+import { safeDecodeURI } from '@wordpress/url';
+
+export const parseAudioUrl = ( src ) => {
+	const decodedURI = safeDecodeURI( src );
+	const fileName = decodedURI
+		.split( '#' )
+		.shift()
+		.split( '?' )
+		.shift()
+		.split( '/' )
+		.pop();
```

Comment:
I think we could remove some of the transformations if we use the [`getPath` function from `@wordpress/url`](https://github.com/WordPress/gutenberg/blob/trunk/packages/url/README.md#getPath), wdyt?

**Resulting code:**
```
const fileName = getPath( safeDecodeURI( src ) ).split( '/' ).pop();
```

Reply:
Good observation 462fd4f